app/src/test/java/com/example/nvdovin/weatherapp/domain/service/WeatherDataServiceTest.java

```diff
@@ -0,0 +1,123 @@
+package com.example.nvdovin.weatherapp.domain.service;
+
+import android.content.Context;
+
+import com.example.nvdovin.weatherapp.data.dao.DaoSession;
+import com.example.nvdovin.weatherapp.data.dao.WeatherDataDao;
+import com.example.nvdovin.weatherapp.data.model.WeatherData;
+
+import org.greenrobot.greendao.Property;
+import org.greenrobot.greendao.query.QueryBuilder;
+import org.greenrobot.greendao.query.WhereCondition;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import java.util.List;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyInt;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class WeatherDataServiceTest {
+
+    private final static Long MOCK_LONG_VALUE = 123L;
+    @Mock
+    Context context;
+    @Mock
+    DaoSession daoSession;
+    //-----------------
+    @Mock
+    List<WeatherData> weatherDataList;
+    @Mock
+    WeatherDataDao weatherDataDao;
+    @Mock
+    QueryBuilder<WeatherData> queryBuilder;
+    @Mock
+    WeatherData weatherData;
+    private WeatherDataService weatherDataService;
+
+    @Before
+    public void setUp() throws Exception {
+        weatherDataService = new WeatherDataService(daoSession, context);
+
+        mockMethods();
+    }
+
+    @Test
+    public void insert() throws Exception {
+
+        weatherDataService.insert(weatherDataList);
+
+        verify(daoSession).getWeatherDataDao();
```

Comment:
You test method insert(), but verify only that getWeatherDataDao() was called. What if someone will change the insert() method and instead of insertOrReplaceInTx(...) will call deleteInTx(...)?